lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -163,41 +164,63 @@ def setUp(self):
     def test_basic(self):
         """Test function returns an array"""
         result = self.dummy_plugin._advect_field(
-            self.data, self.grid_vel_x, self.grid_vel_y, self.timestep, 0.)
-        self.assertIsInstance(result, np.ndarray)
+            self.data, self.grid_vel_x, self.grid_vel_y, self.timestep)
+        self.assertIsInstance(result, np.ma.MaskedArray)
 
     def test_advect_integer_grid_point(self):
         """Test data is advected correctly (by 1 and 2 grid points along the x
         and y axes respectively)"""
-        expected_output = np.array([[0., 0., 0.],
-                                    [0., 0., 0.],
-                                    [0., 2., 3.],
-                                    [0., 1., 2.]])
+        expected_output = np.array([[np.nan, np.nan, np.nan],
+                                    [np.nan, np.nan, np.nan],
+                                    [np.nan, 2., 3.],
+                                    [np.nan, 1., 2.]])
         result = self.dummy_plugin._advect_field(
-            self.data, self.grid_vel_x, self.grid_vel_y, self.timestep, 0.)
-        self.assertArrayAlmostEqual(result, expected_output)
+            self.data, self.grid_vel_x, self.grid_vel_y, self.timestep)
+        self.assertArrayAlmostEqual(result[~result.mask],
+                                    expected_output[~result.mask])
```

Comment:
Why, in this case and all the subsequent ones, are you only checking that 4/6/some sample of values are correct?

I can see that implicitly it means the masking is correct on the returned cube, so we can interpret that there must be nans at the other points. However, could we make the tests easier to understand by just comparing ```result.data.data``` against the ```expected_output``` array? And then compare the ```result.data.mask``` against ```~np.isfinite(expected_output)```.

That very explicitly checks the data and the mask are correct, requiring less interpretation when someone comes back to look at this.

Reply:
Ah I had another thought: I could use:
```
self.assertTrue(np.allclose(result.data.data, expected_data, equal_nan=True))
```
A little non-standard but removes the mask / unmask confusion.  @bayliffe what's your preference?

Reply:
I must have missed the other test. Leave them as is if Stephen is happy.